src/Composer/Installer.php

```diff
@@ -0,0 +1,220 @@
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\ManagerPlugin\Composer;
+
+use Composer\Package\Locker;
+use Contao\ManagerPlugin\Dependency\DependencyResolverTrait;
+use Contao\ManagerPlugin\Dependency\DependentPluginInterface;
+
+class Installer
+{
+    use DependencyResolverTrait;
+
+    /**
+     * @var string
+     */
+    private static $generatedClassTemplate = <<<'PHP'
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\ManagerPlugin;
+
+use Contao\ManagerPlugin\Bundle\BundlePluginInterface;
+use Contao\ManagerPlugin\Config\ConfigPluginInterface;
+use Contao\ManagerPlugin\Config\ExtensionPluginInterface;
+use Contao\ManagerPlugin\Routing\RoutingPluginInterface;
+
+/**
+ * This class has been auto-generated. It will be overwritten at every run of
+ * `composer install` or `composer update`.
+ * 
+ * @see \Contao\ManagerBundle\Composer\Installer
+ */
+%s
+{
+    public const BUNDLE_PLUGINS = BundlePluginInterface::class;
+    public const CONFIG_PLUGINS = ConfigPluginInterface::class;
+    public const EXTENSION_PLUGINS = ExtensionPluginInterface::class;
+    public const ROUTING_PLUGINS = RoutingPluginInterface::class;
+
+    /**
+     * @var array
+     */
+    private $plugins;
+
+    /**
+     * @var array
+     */
+    private $disabled = [];
+
+    public function __construct()
+    {
+        $this->plugins = %s;
+    }
+
+    /**
+     * Returns all active plugin instances.
+     *
+     * @return array
+     */
+    public function getInstances()
+    {
+        return array_diff_key($this->plugins, array_flip($this->disabled));
+    }
+
+    /**
+     * Returns the active plugin instances of a given type (see class constants).
+     *
+     * @param string $type
+     * @param bool   $reverseOrder
+     *
+     * @return array
+     */
+    public function getInstancesOf($type, $reverseOrder = false)
+    {
+        $plugins = array_filter(
+            $this->getInstances(),
+            function ($plugin) use ($type) {
+                return is_a($plugin, $type);
+            }
+        );
+
+        if ($reverseOrder) {
+            $plugins = array_reverse($plugins, true);
+        }
+
+        return array_diff_key($plugins, array_flip($this->disabled));
+    }
+
+    /**
+     * Gets the list of disabled Composer packages.
+     *
+     * @return array
+     */
+    public function getDisabledPackages()
+    {
+        return $this->disabled;
+    }
+
+    /**
+     * Sets the list of disabled Composer packages.
+     *
+     * @param array $packages
+     */
+    public function setDisabledPackages(array $packages)
+    {
+        $this->disabled = $packages;
+    }
+}
+
+PHP;
+
+    /**
+     * Sets the Contao Manager plugins.
+     *
+     * @param Locker $locker
+     *
+     * @throws \RuntimeException
+     */
+    public function dumpPlugins(Locker $locker): void
```

Comment:
I'm not sure about this. This is now using the `Locker` (so `composer.lock`) whereas before we used `installed.json` which would map to the `LocalRepository` (I think). So this is certainly a change but not sure if it makes any difference. In fact they should both contain the packages that are installed. But when I say this it sounds like you should rather use the local repository.

Reply:
changed in latest PR updates

Reply:
changed in 3902fea30435adb86649d3e0d027fbba9dec94f5